cmd/deploy/specs.go

```diff
@@ -130,20 +157,83 @@ func (k *KindSpec) Deploy(ctx context.Context) error {
 	if k.Kubecfg != "" {
 		args = append(args, "--kubeconfig", k.Kubecfg)
 	}
-	if err := k.execer("kind", args...); err != nil {
+	if err := k.execer.exec("kind", args...); err != nil {
 		return errors.Wrap(err, "failed to create cluster using cli")
 	}
 	log.Infof("Deployed kind cluster: %s", k.Name)
+	return k.setupGoogleArtifactRegistryAccess()
+}
+
+func (k *KindSpec) setupGoogleArtifactRegistryAccess() error {
+	if len(k.GoogleArtifactRegistries) == 0 {
+		log.Debug("No registries require setup")
+		return nil
+	}
+	tempDockerDir, err := os.MkdirTemp("", "kne_kind_docker")
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(tempDockerDir)
+	originalConfig := os.Getenv(dockerConfigEnvVar)
```

Comment:
i would maybe suggest a docstring here on why and what you are doing here - it probably isn't super obvious why this is necessary

Reply:
done